### PR TITLE
rework connection limit logic, fixes issue with out of order checks

### PR DESCRIPTION
**Pull request by Greg-Griffith**

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -770,6 +770,23 @@ bool AppInit2(Config &config, thread_group &threadGroup)
...
+
+    // make outbound conns modifiable by the user
+    int nUserMaxOutConnections = GetArg("-maxoutconnections", DEFAULT_MAX_OUTBOUND_CONNECTIONS);
+    nMaxOutConnections = std::max(nUserMaxOutConnections, 0);
+    if (nMaxConnections < nMaxOutConnections)
+    {
+        // uiInterface.ThreadSafeMessageBox((strprintf(_("Reducing -maxoutconnections from %d to %d, because this value
+        // is higher than max available connections."), nUserMaxOutConnections, nMaxConnections)),"",
+        // CClientUIInterface::MSG_WARNING);
+        LOGA(
```
Comment (ptschip) [inline src/init.cpp:781]: Do you still need the commented out code ?

Comment (Greg-Griffith) [reply src/init.cpp:781]: i didnt comment that out. but it doesnt look like it unless we want a warning to pop up to users who are using qt gui

Comment (Greg-Griffith) [reply src/init.cpp:781]: removed

File: src/init.cpp [hunk 2]
```diff
@@ -954,7 +969,7 @@ bool AppInit2(Config &config, thread_group &threadGroup)
     LOGA("Default data directory %s\n", GetDefaultDataDir().string());
     LOGA("Using data directory %s\n", strDataDir);
     LOGA("Using config file %s\n", GetConfigFile(GetArg("-conf", BITCOIN_CONF_FILENAME)).string());
-    LOGA("Using at most %i connections (%i file descriptors available)\n", nMaxConnections, nFD);
+    LOGA("Using at most %i connections\n", nMaxConnections, nFD);
```
Comment (dagurval) [inline src/init.cpp]: You forgot to remove the `nFD` parameter

Comment (sickpig) [reply src/init.cpp]: good catch.

shouldn't we raise an error at compile time in this cases? seems like the LOGA macro is silently accepting more than needed parameters.

Comment (Greg-Griffith) [reply src/init.cpp]: woops. fixed

**Review by sickpig — APPROVED** [review]

**Pull request merged**